metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/GrokController.java

```diff
@@ -53,4 +54,11 @@
     ResponseEntity<Map<String, String>> list() throws RestException {
         return new ResponseEntity<>(grokService.getCommonGrokPatterns(), HttpStatus.OK);
     }
+
+    @ApiOperation(value = "Retrieves a Grok statement from the classpath")
+    @ApiResponse(message = "Grok statement", code = 200)
+    @RequestMapping(value = "/get/statement", method = RequestMethod.GET)
```

Comment:
We have a '/api/v1/hdfs/{path}" endpoint that returns hdfs file contents.  That's the first call the client side makes.  If that fails, it calls '/api/v1/grok/get/statement' to get the pattern from the classpath.  If that fails it reports an error in the UI.

We could change '/api/v1/grok/get/statement' to check HDFS first and change the client side logic to only call the grok endpoint.  Does that make more sense to you and do you think it's worth making that change?